jwst/flatfield/flat_field.py

```diff
@@ -153,20 +153,20 @@ def apply_flat_field(science, flat):
         flat_err = sub_flat.err.copy()
         sub_flat.close()
 
-    # Find pixels in the flat that have a value of NaN and set
-    # their DQ to NO_FLAT_FIELD
+    # Find pixels in the flat that have a value of NaN and add to
+    # DQ to DO_NOT_USE + NO_FLAT_FIELD
+    bad_flag = dqflags.pixel['DO_NOT_USE'] + dqflags.pixel['NO_FLAT_FIELD']
```

Comment:
Probably we can take this out once the flat field reference files are made correctly. There is no flagging of DO_NOT_USE. But it does not harm - if the flat field is a 0 or a NaN it should not be use by the code - this is just a second level of checking.

Reply:
Sometimes you just gotta do what you gotta do, in order to cover all bases, just in case someone else didn't do what they gotta do.

Reply:
My only worry is that because the code this, it won't be obvious if incorrect reference files keep getting delivered.  We won't know.  And given the complexity and the shear number of instances of this in the `flatfield` code, since we are addressing all of these here and now, we will have to go back undo all of this if we want to verify that we have correct reffiles.

Reply:
See discussion here

https://github.com/spacetelescope/jwst/issues/4157#issuecomment-680030042

https://github.com/spacetelescope/jwst/issues/4157#issuecomment-680064961

Reply:
Well I put the 2nd level of checks for that exact reason. Actually the checks were already there but used a different DQ flag.  As someone who will be making reference files during commissioning I think it is going to be a stressful and crazy time. I could see some essential flagging not being set correctly. But it would only add to the stress if the pipeline failed - The pipeline should be smart enough to know that a flat value of 0 of NAN should not be used. The NIRSpec team is going to redeliver reference files but that might take a while.  Having a common sense checks in the pipeline seems ok to me.  - That is my two cents